### PR TITLE
Change unit of log_e_reco from GeV to TeV

### PR DESCRIPTION
**Pull request by rlopezcoto**
Solves #320

### CODE REVIEW

**Comment by rlopezcoto** [comment]
that is going to make a bit messy with the models already produced when applied with newer lstchain versions, right?

My proposal was to change only the `log_e_reco` written in the DL2 file so that it is consistent with the unit of `e_reco`

**Comment by vuillaut** [comment]
> that is going to make a bit messy with the models already produced when applied with newer lstchain versions, right?
> 
> My proposal was to change only the `log_e_reco` written in the DL2 file so that it is consistent with the unit of `e_reco`

Yes it will, but models should not be used between two main releases of lstchain anyway.
So we will just produce new models and IRF with the new release.
Having an inconsistency between `mc` and `reco` will be even worse than the current situation in my opinion.

**Comment by rlopezcoto** [comment]
ok, thanks @vuillaut! 
So we should keep this in mind when re-processing new data (shall the next one be a major release then?)

**Comment by vuillaut** [comment]
> ok, thanks @vuillaut!
> So we should keep this in mind when re-processing new data (shall the next one be a major release then?)

Yes, I think it should - we have introduced major changes recently, no?

I think there are more places where changes should be introduced concerning this PR, for example (but not exhaustive):
- filtering on `log_reco_energy` in build_models by default should now be `-np.inf`
- some labels in plots

File: lstchain/visualization/plot_dl2.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def plot_e(data, n_bins, emin, emax, true_hadroness=False):
     plt.plot(bin_centers,standard_deviations,
              marker='X',linestyle='None')
     plt.ylabel('STD',fontsize=24)
-    plt.xlabel('$log_{10}E_{true}(GeV)$',fontsize=24)
+    plt.xlabel('$log_{10}E_{true}[TeV]$',fontsize=24)
```
Comment (MaxNoe) [inline lstchain/visualization/plot_dl2.py:203]: Should be `r'$\log_{10}(E_\mathrm{true} / TeV)$`

File: lstchain/visualization/plot_dl2.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def plot_features(data, true_hadroness=False):
              histtype=u'step',bins=100,
              label="Protons")
     plt.ylabel(r'# of events',fontsize=15)
-    plt.xlabel(r"$log_{10}E$(GeV)")
+    plt.xlabel(r"$log_{10}E$[TeV]")
```
Comment (MaxNoe) [inline lstchain/visualization/plot_dl2.py:52]: Should be `r'$\log_{10}(E_\mathrm{true} / TeV)$'`

File: lstchain/visualization/plot_dl2.py [hunk 3]
```diff
@@ -432,7 +432,7 @@ def plot_e_resolution(data, n_bins, emin, emax):
     plot1 = ax1.plot(bin_centers,standard_deviations,
                      marker='X',linestyle='None')
     plt.ylabel('STD',fontsize=24)
-    plt.xlabel('$log_{10}E_{true}(GeV)$',fontsize=24)
+    plt.xlabel('$log_{10}E_{true}[TeV]$',fontsize=24)
```
Comment (MaxNoe) [inline lstchain/visualization/plot_dl2.py:435]: see above

File: lstchain/visualization/plot_dl2.py (hunk 2 above)
Comment (rlopezcoto) [reply lstchain/visualization/plot_dl2.py:52]: done

File: lstchain/visualization/plot_dl2.py (hunk 1 above)
Comment (rlopezcoto) [reply lstchain/visualization/plot_dl2.py:203]: done

File: lstchain/visualization/plot_dl2.py (hunk 3 above)
Comment (rlopezcoto) [reply lstchain/visualization/plot_dl2.py:435]: done

File: lstchain/reco/dl0_to_dl1.py [hunk 4]
```diff
@@ -344,7 +344,7 @@ def r0_to_dl1(input_filename = get_dataset_path('gamma_test_large.simtel.gz'),
                     # Log10(Energy) in GeV
                     if is_simu:
                         dl1_container.mc_energy = event.mc.energy.value
-                        dl1_container.log_mc_energy = np.log10(event.mc.energy.value * 1e3)
+                        dl1_container.log_mc_energy = np.log10(event.mc.energy.value)
```
[the remark this answers is not in the document]
Comment (rlopezcoto) [reply lstchain/reco/dl0_to_dl1.py:347]: changed

[the remark this answers is not in the document]
Comment (MaxNoe) [reply lstchain/reco/dl0_to_dl1.py:347]: I don't see it yet here, forgot to push?

File: lstchain/visualization/plot_dl2.py (hunk 3 above)
Comment (MaxNoe) [reply lstchain/visualization/plot_dl2.py:435]: also not seeing this yet

**Comment by MaxNoe** [comment]
@vuillaut @rlopezcoto there were no new commits since I reviewed, did you forget to push?

**Comment by vuillaut** [comment]
> @vuillaut @rlopezcoto there were no new commits since I reviewed, did you forget to push?

I saw that @rlopezcoto said "done" to each comment and assumed (too quickly) they have been addressed.

**Comment by rlopezcoto** [comment]
sorry, weird... I was left with no space on my computer and files were modified but commit/push did not work...

**Pull request merged**